src/map/skill.cpp

```diff
@@ -5632,7 +5632,7 @@ int skill_castend_damage_id (struct block_list* src, struct block_list *bl, uint
 		}
 		break;
 	case GC_CROSSIMPACT:
-		if (skill_check_unit_movepos(5, src, bl->x, bl->y, 1, 1))
+		if (skill_check_unit_movepos(5, src, bl->x, bl->y, 1, 1)) // !TODO: Confirm cast distance
```

Comment:
what about replace it with 
	case GC_CROSSIMPACT:
		{
			short x, y;
			short dir = map_calc_dir(src,bl->x,bl->y);

			if( dir > 0 && dir < 4) x = 1;
			else if( dir > 4 ) x = -1;
			else x = 0;
			if( dir > 2 && dir < 6 ) y = 1;
			else if( dir == 7 || dir < 2 ) y = -1;
			else y = 0;

			if( unit_movepos(src, bl->x+x, bl->y+y, 1, 1) ) {
				clif_blown(src);
				skill_attack(BF_WEAPON,src,src,bl,skill_id,skill_lv,tick,flag);
			}

		}
		break;
?

Reply:
I guess what I'd like is a video or GIF of it's behavior on kRO.

Reply:
![ROAOJTJ7E8LBJJQV0WLY](https://user-images.githubusercontent.com/54556365/76787871-7efc4600-67c2-11ea-98a8-eade6a88b5c6.gif)
i think this gif can helps you.
there is more videos

Reply:
I'd like to have something of a cast at a greater difference. The description says that the caster will move the target immediately which either is a misinterpretation of "walk to target" or means "teleport to target".

Reply:
in gif you can see that it teleports to the target, look at the moment after poring is dead. when he uses cross impact he gets teleported to the egg.

Reply:
Right. The current implementation behaves like this.

Reply:
![GIF 16-May-20 6-30-55 PM](https://user-images.githubusercontent.com/9781158/82118750-749b0000-97a3-11ea-84fe-2564a6777f05.gif)
It does not jump to the side of the monster